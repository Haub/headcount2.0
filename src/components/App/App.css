.headcount-title {
  text-align: center;
}
