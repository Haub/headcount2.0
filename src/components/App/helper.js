import kinderData from '../../data/kindergartners_in_full_day_program';

export default class DistrictRepository {
  
  constructor(data) {
    this.stats = this.removeDuplicates(kinderData)
  }

  removeDuplicates = (kinderData) => {
    return kinderData.reduce((kinderObj, district) => {
      let schoolDistrict = district.Location;
      let year = district.TimeFrame;
      let roundedData = Math.round(district.Data * 1000)/1000;
      
      if(!kinderObj[schoolDistrict]) {
        kinderObj[schoolDistrict] = {
          location: schoolDistrict, 
          stats: {}
        }
      } 
      
      kinderObj[schoolDistrict].stats[year] = roundedData || 0;
      return kinderObj
    
    }, {})  
  }

  findByName = (schoolDistrict = '') => {
    const upperSchoolDistrict = schoolDistrict.toUpperCase();
    return this.stats[upperSchoolDistrict] 
  }



  findAllMatches = (schoolDistrict = '') => {
    const districts = Object.keys(this.stats);
    const matchingDistricts = districts.filter(district => district.toUpperCase().includes(schoolDistrict.toUpperCase()));
    return matchingDistricts
  }
}