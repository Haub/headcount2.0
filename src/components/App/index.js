import React, { Component } from 'react';
import './App.css';
import DistrictRepository from './helper';
import kinderData from '../../data/kindergartners_in_full_day_program';
import { CardContainer } from '../CardContainer';
import Search from '../Search';

class App extends Component {
  constructor() {
    super();
    this.state = {
      district: new DistrictRepository(kinderData),
      filteredDistricts: []
    };
  }
  
  componentDidMount = () => {
    this.setState({filteredDistricts: this.state.district.stats});
  }

  filterDistricts = (districtInput) => {
    const filteredDistricts = this.state.district.findAllMatches(districtInput);
    this.setState({ filteredDistricts });
  }

  render() {
    return (
      <div>
        <header>
          <h1 className='headcount-title'>Headcount 2.0</h1>
          <Search filterDistricts={this.filterDistricts} />
        </header>
        <CardContainer filteredDistricts={this.state.filteredDistricts} />
      </div>
    );
  }
}

export default App;
