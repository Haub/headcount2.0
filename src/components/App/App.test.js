import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './index';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sets the state of app with all districts in componentDidMount', () => {
    wrapper.instance().componentDidMount();
    expect(wrapper.state().filteredDistricts).toHaveLength(181);
  });

  it('filters districts based on user input', () => {
    wrapper.instance().filterDistricts('colorado');
    expect(wrapper.state().filteredDistricts).toHaveLength(2);
  });

  
});
