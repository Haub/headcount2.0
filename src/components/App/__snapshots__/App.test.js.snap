// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1
                    className="headcount-title"
          >
                    Headcount 2.0
          </h1>
          <Search
                    filterDistricts={[Function]}
          />
</header>,
        <Unknown
          filteredDistricts={Array []}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="headcount-title"
>
              Headcount 2.0
</h1>,
            <Search
              filterDistricts={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Headcount 2.0",
              "className": "headcount-title",
            },
            "ref": null,
            "rendered": "Headcount 2.0",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "filterDistricts": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "filteredDistricts": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
                        className="headcount-title"
            >
                        Headcount 2.0
            </h1>
            <Search
                        filterDistricts={[Function]}
            />
</header>,
          <Unknown
            filteredDistricts={Array []}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="headcount-title"
>
                Headcount 2.0
</h1>,
              <Search
                filterDistricts={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Headcount 2.0",
                "className": "headcount-title",
              },
              "ref": null,
              "rendered": "Headcount 2.0",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "filterDistricts": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "filteredDistricts": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
